### PR TITLE
fix(migration): ensure to have only one dataset per attribute instance

### PR DESCRIPTION
**Pull request by gpoujol**

### CODE REVIEW

File: search-service/src/test/resources/ngsild/entities_with_two_instances_who_have_same_dataset_id.jsonld [hunk 1]
```diff
@@ -0,0 +1,857 @@
+{
```
Comment (bobeal) [inline search-service/src/test/resources/ngsild/entities_with_two_instances_who_have_same_dataset_id.jsonld:1]: do you **really** need such a laaarge payload?

File: search-service/src/test/kotlin/db/migration/VO_28_JsonLd_migrationTests.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@ActiveProfiles("test")
+class VO_28_JsonLd_migrationTests {
+
+    @Test
+    fun `it should remove instances when attribute have more than one instance with the same datasetId`() {
+        val payload =
+            loadSampleData("entities_with_two_instances_who_have_same_dataset_id.jsonld").deserializeAsMap()
+        val cleanPayload = payload.keepOnlyOneInstanceByDatsetId()
+
+        assertNotEquals(payload, cleanPayload)
```
Comment (bobeal) [inline search-service/src/test/kotlin/db/migration/VO_28_JsonLd_migrationTests.kt:18]: a bit weak

Comment (bobeal) [reply search-service/src/test/kotlin/db/migration/VO_28_JsonLd_migrationTests.kt:18]: better to compare with an expected payload (easier if you don't add a laaaarge test payload)

File: search-service/src/test/resources/ngsild/entities_with_two_instances_who_have_same_dataset_id.jsonld (hunk 1 above)
Comment (gpoujol) [reply search-service/src/test/resources/ngsild/entities_with_two_instances_who_have_same_dataset_id.jsonld:1]: No, I can just leave what we need

**Comment by bobeal** [comment]
there is again a base branch problem, there are in this PR commits that should not be there

File: search-service/src/test/kotlin/db/migration/VO_28_JsonLd_migrationTests.kt (hunk 2 above)
Comment (gpoujol) [reply search-service/src/test/kotlin/db/migration/VO_28_JsonLd_migrationTests.kt:18]: Ok I change this

Comment (gpoujol) [reply search-service/src/test/kotlin/db/migration/VO_28_JsonLd_migrationTests.kt:18]: I base my branch on the PR that I just merged that's why

**Comment by gpoujol** [comment]
> there is again a base branch problem, there are in this PR commits that should not be there

I base my branch on the PR that I just merged that's why

**Comment by bobeal** [comment]
> > there is again a base branch problem, there are in this PR commits that should not be there
> 
> I base my branch on the PR that I just merged that's why

if your PR does not depend on the other PR (and it does not), do it directly on top of the target branch.

**Comment by bobeal** [comment]
Just pushed a commit with a lot of fixes and improvements.

**Comment by gpoujol** [comment]
It's good to me. Can I merge it ? @bobeal

**Review by bobeal — APPROVED** [review]

**Pull request merged**